File: nexobytes_edutech/lib/classes/subject.dart
class Subject {
  String id;
  int createdAt;
  String classId;
  String className;
  String createdBy;
  String name;

  Subject({
    required this.id,
    required this.name,
    this.createdBy="",
    this.createdAt=0,
    required this.classId,
    this.className=""
  });
}

File: server/models/subject.js
const mongoose = require("mongoose");
const uniqueValidator = require('mongoose-unique-validator');

const Schema = mongoose.Schema;

const subjectSchema = new Schema({
	subjectId: { type: String, required: true, unique: true },
	classId: { type: String, required: true },
	className: { type: String, required: true },
	name: { type: String, required: true },
	createdAt: { type: Number, required: true },
	createdBy: { type: String, required: true },
});

subjectSchema.plugin(uniqueValidator);

module.exports = mongoose.model("Subject", subjectSchema);
